### PR TITLE
Multiplayer: Add Viewport Message When Users Incorrectly Load Level

### LINKED ISSUE #9666: Multiplayer: As a Dev I want a Warning When Loading Network Levels At Bad Times Depending on Multiplayer Status
**Is your feature request related to a problem? Please describe.**
The following workflow won't work, but is very easy to mistake as possible. As a dev, I want a warning whenever attempting to do the following...
_Server > loadlevel > host_

The reason this order of operation doesn't work is that when loading a networked level, the Multiplayer System still is uninitialized (ie the app doesn't know if it's a client, dedicatedserver, or clientserver).

**Describe the solution you'd like**
Check whenever a prefab (level) is loaded, and if that prefab is associated with a network.spawnable then check if the IMultiplayer system is initialized to be a client or server. If it's uninitialized, then display a prominent warning.

**Describe alternatives you've considered**
It might be good even to block loading the prefab entirely. Why would you load a network spawnable if there's no network?

### LINKED ISSUE #11751: Bug Report: Client server connection test is hard to read as its white text against a mostly white environment.
**Describe the bug**
I am a big fan of the new connection text for multiplayer. However the whitetext against a white environment is hard to read. 

**Assets required**
[!Important!] Please provide sample assets needed to reproduce the issue, either as an [attachment](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files) or as a link to a public asset in Github.

**Steps to reproduce**
1. Build a Multiplayer project like Automated Testing or Multiplayer Sample
2. Go to play in editor
3. Look at the multiplayer debug connection text which covers the three major connection steps: launching the server, connecting to the server and serializaing data.

**Expected behavior**
* Text is easy to read and visually distinguishable from background.

**Actual behavior**
* White text on white background makes text hard to parse (would consider this an accessibility problem).
capture1

**Screenshots/Video**
If applicable, add screenshots and/or a video to help explain your problem.

**Found in Branch**
Development

**Desktop/Device (please complete the following information):**
 - Device: PC 
 - OS: Windows

**Additional context**

### PR DESCRIPTION
**Pull request by AMZN-Gene**
1. Add a on-screen toast to the viewport when users try to load a level that's blocked by the multiplayer system. 
2. Adds a scrim (mild background gradient) when multiplayer debug text is up to add contrast and thus improve readability
image

_Before_
image

_After_
image

Signed-off-by: Gene Walters <genewalt@amazon.com>

Resolves #9666 
Resolves #11751

## How was this PR tested?
Running the editor and launchers to see blocked level loading and scrim appears and hides itself.

### CODE REVIEW

**Comment by rgba16f** [comment]
Is there a reason to use the SetState rather than the individual accessor functions?

File: Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp [hunk 1]
```diff
@@ -252,6 +370,45 @@ namespace Multiplayer
...
+        AzFramework::DebugDisplayRequestBus::Bind(debugDisplayBus, viewport->GetId());
+        AzFramework::DebugDisplayRequests*  debugDisplay = AzFramework::DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus);
+        if (!debugDisplay)
+        {
+            return;
+        }
+
+        const AZ::u32 previousState = debugDisplay->GetState();
+        debugDisplay->SetState(
+            e_Mode2D | e_AlphaBlended | e_FillModeSolid | e_CullModeBack | e_DepthWriteOff | static_cast<AZ::u32>(e_DepthTestOff));
```
Comment (rgba16f) [inline Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp:391]: Other than 2d mode these can all be done using the direct access functions. 2D mode would require a Draw2DQuadGradient function.
```c++
debugDisplay->SetAlpha(ScrimAlpha * alphaMultiplier);
// fill mode unsupported
debugDisplay->CullOn();
debugDisplay->DepthWriteOff();
debugDisplay->DepthTestOff();
```

Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp:391]: That would be nice. In other words, add a new DebugDisplayRequests method: DrawQuadGradient2d?

Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp]: Added DrawQuad2dGradient to DebugDisplayRequests. Let me know what you think!

File: Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp [hunk 2]
```diff
@@ -386,27 +316,25 @@ namespace Multiplayer
             return;
         }
 
-        const AZ::u32 previousState = debugDisplay->GetState();
```
Comment (rgba16f) [inline Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp]: Ironically you probably still want to store and restore the previous state :)

We need to update the API to have a push state & pop state functions to make this straightforward, but that can be done later.

**Review by rgba16f — APPROVED** [review]
Change looks good to me, just needs to store and restore the render state.

File: Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp (hunk 2 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp]: Good catch. The call to DrawQuad doesn't affect the state, but debugDisplay->DepthWriteOff() and DepthTestOff() does.
 Fixed!
